src/bin/ion/commands/schema/validate.rs

```diff
@@ -113,46 +106,27 @@ impl IonCliCommand for ValidateCommand {
             .with_context(|| format!("Schema {} does not have type {}", schema_id, schema_type))?;
 
         // create a text writer to make the output
-        let mut output = vec![];
-        let mut writer = TextWriterBuilder::new(TextKind::Pretty).build(&mut output)?;
+        let mut writer = Writer::new(v1_0::Text.with_format(TextFormat::Pretty), vec![])?;
 
         // validate owned_elements according to type_ref
-        for owned_element in owned_elements {
+        for owned_element in elements {
             // create a validation report with validation result, value, schema and/or violation
-            writer.step_in(IonType::Struct)?;
+            let mut struct_writer = writer.struct_writer()?;
             let validation_result = type_ref.validate(&owned_element);
-            writer.set_field_name("result");
             match validation_result {
                 Ok(_) => {
-                    writer.write_string("Valid")?;
-                    writer.set_field_name("value");
-                    writer.write_string(element_to_string(&owned_element)?)?;
-                    writer.set_field_name("schema");
-                    writer.write_string(schema_id)?;
+                    struct_writer.write("result", "Valid")?;
+                    struct_writer.write("value", format!("{}", &owned_element))?;
+                    struct_writer.write("schema", schema_id)?;
```

Comment:
`struct_writer` really cleans this up, nice :)

Reply:
Yes, @zslayton put a lot of effort into trying different approaches in order to find an ergonomic writer API.